### ISSUE TITLE
Per-Table HTTP Connection Pooling Behavior

### ISSUE THREAD

**Issue by robby-bryant** [NerdWalletOSS/dynamorm#49]
**tl;dr** - DynamORM inadvertently any given Dynamo table to 10 concurrent HTTP connections.

**Details**:
- Dynamo Operations
    - `DynamoTable3` ultimately performs all Dynamo operations via `.get_resource().Table(name).foo(...)` which delegates to the underlying boto table operations.  
    - boto table operations are ultimately executed by a Client that is dynamically generated by `botocore.client.ClientCreator`.  This factory generates a set of `botocore.endpoint.Endpoint` objects, which use an underlying `botocore.endpoint.BotocoreHTTPSession` adapter to the Python `requests` library.
    - The `botocore.endpoint.BotocoreHTTPSession` manages a `requests.Session` object, which is configured to manage an HTTP connection pool.  The default max pool size is 10.
- Dynamo Resource Manager
    - `DynamoTable3.get_resource` creates an underlying `boto3.Session` and then a resource via `boto3.Session.resource`.  This `boto3.Session.resource` call is what creates the underlying Client object for performing Dynamo operations.
    - `DynamoTable3.get_resource` caches the resource for later use.
    - Both `boto3.Session.resource` and then `boto3.Session.client`accept an optional `botocore.config.Config` keyword argument that has a `max_pool_connections` field that is threaded down to the `requests.Session` connection pool.
- Putting it All Together
    - Each Dynamo table manages a single cached boto Resource, which manages a single Client, which manages a single `botocore.endpoint.HTTPSession`, which manages a single `requests.Session` with a default max pool size of 10.

**Proposed Solutions**:
- Solution 1: Clients that need more than 10 concurrent connections to a single Dynamo endpoint can configure the max pool size by passing in `resource_kwargs` arguments to Dynamodel Table definition.
    - Issue: The underling config kwarg that `botocore.Session.resource` expects needs to be an actual `botocore.config.Config` instance, but `DynamoTable3.get_resource` does a JSON dump of all the table config to generate a cache key, and the `Config` instance is not JSON serializable.
        - Possible Workaround: `DynamoTable3` looks out for any `config` key in `resource_kwargs`' and automatically wraps it in a `Config` object after JSON serialization.
        - Possible Workaround: max pool size becomes a first class citizen in the Table definition and is handled explicitly in `get_resource`.
- Solution 2: We abandon resource caching and serve up a fresh resource object (with fresh `requests.Session` connection pool) every time.
    - Issue: Full table definition code is required for boto Session and Resource creation, but Dynamorm code only passes full table definition config to get_resource during initial table management operations.  During steady-state operations, this (required) config is omitted, but the resource is served out of cache, so everything works out smoothly. 
        - Possible workaround: Automatically inject full table config on all `get_resource` calls.
    - Issue: We don't know what the performance impact of per-operation Session and Resource setup will be.
